Sources/ContainerClient/Core/Volume.swift

```diff
@@ -0,0 +1,223 @@
+    case local = "local"
+    case global = "global"
+}
+
+/// Usage information for a volume.
+public struct VolumeUsageData: Sendable, Codable, Equatable {
+    public var size: Int64
+    // Number of containers currently using this volume.
+    public var refCount: Int64
+
+    public init(size: Int64, refCount: Int64) {
+        self.size = size
+        self.refCount = refCount
+    }
+}
+
+/// Request to create a new volume.
+public struct VolumeCreateRequest: Sendable, Codable {
+    public var name: String
+    public var driver: String
+    public var driverOpts: [String: String]
+    public var labels: [String: String]
+
+    public init(
+        name: String,
+        driver: String = "local",
+        driverOpts: [String: String] = [:],
+        labels: [String: String] = [:]
+    ) {
+        self.name = name
+        self.driver = driver
+        self.driverOpts = driverOpts
+        self.labels = labels
+    }
+}
+
+/// Request to delete a volume.
+public struct VolumeDeleteRequest: Sendable, Codable {
+    public var name: String
+    public var force: Bool
+
+    public init(name: String, force: Bool = false) {
+        self.name = name
+        self.force = force
+    }
+}
+
+/// Response containing a list of volumes.
+public struct VolumeListResponse: Sendable, Codable {
+    public var volumes: [Volume]
+    public var warnings: [String]?
+
+    public init(volumes: [Volume], warnings: [String]? = nil) {
+        self.volumes = volumes
+        self.warnings = warnings
+    }
+}
+
+/// Response containing volume details.
+public struct VolumeInspectResponse: Sendable, Codable {
+    public var volume: Volume
+
+    public init(volume: Volume) {
+        self.volume = volume
+    }
+}
+
+/// Error types for volume operations.
+public enum VolumeError: Error, LocalizedError {
+    case volumeNotFound(String)
+    case volumeAlreadyExists(String)
+    case volumeInUse(String)
+    case invalidVolumeName(String)
+    case driverNotSupported(String)
+    case storageError(String)
+
+    public var errorDescription: String? {
+        switch self {
+        case .volumeNotFound(let name):
+            return "Volume '\(name)' not found"
+        case .volumeAlreadyExists(let name):
+            return "Volume '\(name)' already exists"
+        case .volumeInUse(let name):
+            return "Volume '\(name)' is in use and cannot be removed"
+        case .invalidVolumeName(let name):
+            return "Invalid volume name '\(name)'"
+        case .driverNotSupported(let driver):
+            return "Volume driver '\(driver)' is not supported"
+        case .storageError(let message):
+            return "Storage error: \(message)"
+        }
+    }
+}
+
+/// Volume storage management utilities.
+public struct VolumeStorage {
+    public static let volumesDirectory = URL(fileURLWithPath: NSHomeDirectory())
+        .appendingPathComponent("Library")
+        .appendingPathComponent("Application Support")
```

Comment:
```swift
       let root = FileManager.default.urls(
            for: .applicationSupportDirectory,
            in: .userDomainMask
        ).first
```

Reply:
> ```swift
> let root = FileManager.default.urls(
>             for: .applicationSupportDirectory,
>             in: .userDomainMask
>         ).first
> ```

Better: `Application.appRoot` where you need the app data root.